component-api/src/main/java/org/talend/sdk/component/api/service/record/RecordBuilderFactory.java

```diff
@@ -62,4 +65,27 @@
      * @return a builder to create a schema entry.
      */
     Schema.Entry.Builder newEntryBuilder();
+
+    /**
+     * Build a schema.entry from another one. Useful to duplicate a column with some changes.
+     * 
+     * @param model : model of entry to copy.
+     * @return entry builder with model parameters.
+     */
+    default Schema.Entry.Builder newEntryBuilder(Schema.Entry model) {
```

Comment:
Why not... but don't we have enough?

Reply:
It's a facility function, similar to 
```java
Schema.Builder newSchemaBuilder(Schema schema);
Record.Builder newRecordBuilder(Schema schema, Record record);
```
and can help on DataPrep processor, where we can add a new Field that is a copy of an existing one but with a different name.